### PR TITLE
test: ensure client-side invalid state is not propagated to the server

### PR DESCRIPTION
**Pull request by vursen**
## Description

To err on the side of caution, I added integration tests that ensure the invalid state doesn't get propagated to the server when modified on the client-side, which can be a severe vulnerability otherwise. 

Depends on 
- https://github.com/vaadin/flow-components/pull/4342
- https://github.com/vaadin/flow-components/pull/4343

Part of https://github.com/vaadin/platform/issues/3066

## Type of change

- [x] Internal

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/contributing/overview
- [x] I have added a description following the guideline.
- [x] I created the issue in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

### CODE REVIEW

File: vaadin-checkbox-flow-parent/vaadin-checkbox-flow-integration-tests/src/test/java/com/vaadin/flow/component/checkbox/tests/validation/BasicValidationIT.java [hunk 1]
```diff
@@ -69,6 +69,17 @@ public void webComponentCanNotModifyInvalidState() {
         assertWebComponentCanNotModifyInvalidState();
     }
 
+    @Test
+    public void clientSideInvalidStateIsNotPropagatedToServer() {
+        // Make the field invalid
+        $("button").id(REQUIRED_BUTTON).click();
+        testField.sendKeys(Keys.TAB);
```
Comment (sissbruecker) [inline vaadin-checkbox-flow-parent/vaadin-checkbox-flow-integration-tests/src/test/java/com/vaadin/flow/component/checkbox/tests/validation/BasicValidationIT.java:76]: `CheckBoxGroup` requires tabbing from a child checkbox specifically:
https://github.com/vaadin/flow-components/blob/cadfe69f0594feeda0389a8b3ea7385de195602b/vaadin-checkbox-flow-parent/vaadin-checkbox-flow-integration-tests/src/test/java/com/vaadin/flow/component/checkbox/tests/validation/BasicValidationIT.java#L33

File: vaadin-checkbox-flow-parent/vaadin-checkbox-flow-integration-tests/src/test/java/com/vaadin/flow/component/checkbox/tests/validation/BasicValidationIT.java [hunk 2]
```diff
@@ -69,6 +69,17 @@ public void webComponentCanNotModifyInvalidState() {
         assertWebComponentCanNotModifyInvalidState();
     }
 
+    @Test
+    public void clientSideInvalidStateIsNotPropagatedToServer() {
```
Comment (sissbruecker) [inline vaadin-checkbox-flow-parent/vaadin-checkbox-flow-integration-tests/src/test/java/com/vaadin/flow/component/checkbox/tests/validation/BasicValidationIT.java:73]: Can also add this for `RadioButtonGroup`.

File: vaadin-text-field-flow-parent/vaadin-text-field-flow-integration-tests/src/test/java/com/vaadin/flow/component/textfield/tests/validation/PasswordFieldValidationBasicIT.java [hunk 3]
```diff
@@ -68,6 +68,17 @@ public void triggerInputBlur_assertValidity() {
         assertClientValid();
     }
 
+    @Test
+    public void clientSideInvalidStateIsNotPropagatedToServer() {
+        // Make the field invalid
+        $("button").id(REQUIRED_BUTTON).click();
+        testField.sendKeys(Keys.TAB);
```
Comment (sissbruecker) [inline vaadin-text-field-flow-parent/vaadin-text-field-flow-integration-tests/src/test/java/com/vaadin/flow/component/textfield/tests/validation/PasswordFieldValidationBasicIT.java:75]: Requires double tab to pass.

File: vaadin-date-time-picker-flow-parent/vaadin-date-time-picker-flow-integration-tests/src/test/java/com/vaadin/flow/component/datetimepicker/validation/DateTimePickerValidationBasicIT.java [hunk 4]
```diff
@@ -64,6 +64,17 @@ public void detach_attach_onlyServerCanSetFieldToValid() {
         onlyServerCanSetFieldToValid();
     }
 
+    @Test
+    public void clientSideInvalidStateIsNotPropagatedToServer() {
+        // Make the field invalid
+        $("button").id(REQUIRED_BUTTON).click();
+        testField.sendKeys(Keys.TAB);
```
Comment (sissbruecker) [inline vaadin-date-time-picker-flow-parent/vaadin-date-time-picker-flow-integration-tests/src/test/java/com/vaadin/flow/component/datetimepicker/validation/DateTimePickerValidationBasicIT.java:71]: This also needs two tabs apparently:
https://github.com/vaadin/flow-components/blob/cadfe69f0594feeda0389a8b3ea7385de195602b/vaadin-date-time-picker-flow-parent/vaadin-date-time-picker-flow-integration-tests/src/test/java/com/vaadin/flow/component/datetimepicker/validation/DateTimePickerValidationBasicIT.java#L71-L72

Comment (vursen) [reply vaadin-date-time-picker-flow-parent/vaadin-date-time-picker-flow-integration-tests/src/test/java/com/vaadin/flow/component/datetimepicker/validation/DateTimePickerValidationBasicIT.java]: Right, done.

File: vaadin-text-field-flow-parent/vaadin-text-field-flow-integration-tests/src/test/java/com/vaadin/flow/component/textfield/tests/validation/PasswordFieldValidationBasicIT.java (hunk 3 above)
Comment (vursen) [reply vaadin-text-field-flow-parent/vaadin-text-field-flow-integration-tests/src/test/java/com/vaadin/flow/component/textfield/tests/validation/PasswordFieldValidationBasicIT.java:75]: Right, done.

File: vaadin-checkbox-flow-parent/vaadin-checkbox-flow-integration-tests/src/test/java/com/vaadin/flow/component/checkbox/tests/validation/BasicValidationIT.java (hunk 2 above)
Comment (vursen) [reply vaadin-checkbox-flow-parent/vaadin-checkbox-flow-integration-tests/src/test/java/com/vaadin/flow/component/checkbox/tests/validation/BasicValidationIT.java:73]: Right, done.

File: vaadin-checkbox-flow-parent/vaadin-checkbox-flow-integration-tests/src/test/java/com/vaadin/flow/component/checkbox/tests/validation/BasicValidationIT.java (hunk 1 above)
Comment (vursen) [reply vaadin-checkbox-flow-parent/vaadin-checkbox-flow-integration-tests/src/test/java/com/vaadin/flow/component/checkbox/tests/validation/BasicValidationIT.java]: Done.

**Review by sissbruecker — APPROVED** [review]

File: vaadin-text-field-flow-parent/vaadin-text-field-flow-integration-tests/src/test/java/com/vaadin/flow/component/textfield/tests/validation/IntegerFieldValidationBasicIT.java [hunk 5]
```diff
@@ -69,14 +69,14 @@ public void clientSideInvalidStateIsNotPropagatedToServer() {
     }
 
     @Test
-    public void triggerInputBlur_assertValidity() {
+    public void triggerBlur_assertValidity() {
```
Comment (vursen) [inline vaadin-text-field-flow-parent/vaadin-text-field-flow-integration-tests/src/test/java/com/vaadin/flow/component/textfield/tests/validation/IntegerFieldValidationBasicIT.java:72]: **note:** A leftover from https://github.com/vaadin/flow-components/pull/4336. Let me know if you feel it deserves a separate PR.

Comment (vursen) [reply vaadin-text-field-flow-parent/vaadin-text-field-flow-integration-tests/src/test/java/com/vaadin/flow/component/textfield/tests/validation/IntegerFieldValidationBasicIT.java:72]: After more consideration, I decided to create a separate PR containing this as well as a few other changes: https://github.com/vaadin/flow-components/pull/4342
